packages/core/src/tools/create-cache.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const _ = require('lodash');
+
+const JSON = require('./create-json-tool')();
+
+const createCacheObj = (input) => {
+  const rpc = _.get(input, '_zapier.rpc');
+  const runValidationChecks = (rpc, key, value = null, ttl = null) => {
+    if (!rpc) {
+      throw new Error('rpc is not available');
+    }
+
+    if (!_.isString(key)) {
+      throw new Error('key must be a string');
+    }
+
+    if (!JSON.stringify(value)) {
+      throw new Error('value must be JSON-encodable');
+    }
+
+    if (ttl != null && !_.isInteger(ttl)) {
+      throw new Error('ttl must be a number');
+    }
```

Comment:
I think it's okay for `ttl` to be a non-integer number.

```suggestion
    if (ttl != null && typeof ttl !== 'number') {
      throw new TypeError('ttl must be a number');
    }
```

Or if you want to keep `ttl` an integer, we should change the error message:

```js
    if (ttl != null && !_.isInteger(ttl)) {
      throw new TypeError('ttl must be an integer');
    }
```